controllers/argocd/util.go

```diff
@@ -364,7 +364,33 @@ func (r *ReconcileArgoCD) getDexOAuthClientSecret(cr *argoprojv1a1.ArgoCD) (*str
 	}
 
 	if tokenSecret == nil {
```

Comment:
`// This change is done, due to change of reduction of secret-based service account tokens in k8s v1.24 so from k8s v1.24 no default secret for service account, but for dex to work we need to provide token of secret used by dex service account  as a oauth token, this helps to achieve that, in log run we should see do dex really requires a secret or it manages to create one using TokenRequest API
`

Reply:
is this ok @jaideepr97 ?

Reply:
looks fine, just as long as we have something! 
thanks